crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -676,6 +671,317 @@ TEST_P(HMAC_ServiceIndicatorTest, HMACTest) {
   ASSERT_TRUE(check_test(hmacTestVector.expected_digest, mac.data(), mac_len, "HMAC KAT"));
 }
 
+TEST(ServiceIndicatorTest, RSAKeyGen) {
+ int approved = AWSLC_NOT_APPROVED;
+ bssl::UniquePtr<RSA> rsa(RSA_new());
+ ASSERT_TRUE(rsa);
+
+ // RSA_generate_key_fips may only be used for 2048-, 3072-, and 4096-bit
+ // keys.
+ ASSERT_FALSE(RSA_generate_key_fips(rsa.get(), 512, nullptr));
```

Comment:
Do we want to surround these with a check for `Not Approved`? Or that would be an overkill? It can be done like the loop below.